src/ngx_http_lua_headers_in.c

```diff
@@ -683,50 +684,56 @@ lua_req_set_header_hash_init(void)
             goto failed;
         }
         tmp_key.data = &header_string[pos];
-        strcpy((char *)tmp_key.data, (char *)handlers[i].name.data);
+        strncpy((char *)tmp_key.data, (char *)handlers[i].name.data,
```

Comment:
In the nginx world, we usually use the ngx_xxx form. For this case, `ngx_memcpy` should be better.